### PR TITLE
Add yarn support for generated projects

### PR DESCRIPTION
**Pull request by frbuceta**
One of the commits ignores the prompt when checking required in a model.
The other adds the functionality of using different package managers.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by frbuceta** [comment]
The tests are failing because they do not set `packageManager: 'npm'` by default

**Comment by raymondfeng** [comment]
@frbuceta It's nice to support `yarn`. Can you add `yarn` installation to `.travis.yml` and `appveyor.yml` for CI?

**Comment by frbuceta** [comment]
> @frbuceta It's nice to support `yarn`. Can you add `yarn` installation to `.travis.yml` and `appveyor.yml` for CI?
> 
> See https://classic.yarnpkg.com/en/docs/install-ci/

I have already implemented it, as in the guide

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
 env: TASK=test
+
+cache:
+  yarn: true
+
+before_install:
+  - sudo apt-key adv --fetch-keys http://dl.yarnpkg.com/debian/pubkey.gpg
+  - echo "deb http://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+  - sudo apt-get update -qq
+  - sudo apt-get install -y -qq yarn
```
Comment (raymondfeng) [inline .travis.yml:25]: Is there a better way to install `yarn` that does not require `sudo`?

Comment (raymondfeng) [reply .travis.yml:25]: Is `npm i -g yarn` safe?

Comment (frbuceta) [reply .travis.yml:25]: It should be safe, I think it's a good idea

Comment (frbuceta) [reply .travis.yml]: works

File: packages/cli/test/integration/generators/copyright-monorepo.integration.js [hunk 2]
```diff
@@ -38,7 +38,12 @@ describe('lb4 copyright for monorepo', function () {
           additionalFiles: SANDBOX_FILES,
         }),
       )
-      .withOptions({gitOnly: false, owner: 'ACME Inc.', license: 'MIT'});
+      .withOptions({
+        gitOnly: false,
+        owner: 'ACME Inc.',
+        license: 'MIT',
+        packageManager: 'npm', // Default
```
Comment (raymondfeng) [inline packages/cli/test/integration/generators/copyright-monorepo.integration.js:45]: What's going to happen if this option is not set? Does it default to `npm`?

Comment (frbuceta) [reply packages/cli/test/integration/generators/copyright-monorepo.integration.js:45]: I do not know what is happening... The other tests take the default value that is NPM but those do not.
```
➜  loopback-next git:(changes-in-cli) ./packages/cli/bin/cli-main.js app -h
Usage:
  lb4 app [<name>] [options]

Options:
  -h,    --help             # Print the generator's options and usage
         --skip-cache       # Do not remember prompt answers                                Default: false
         --skip-install     # Do not automatically install dependencies                     Default: false
         --force-install    # Fail on install dependencies error                            Default: false
         --ask-answered     # Show prompts for already configured options                   Default: false
         --applicationName  # Application class name
         --docker           # Include Dockerfile and .dockerignore
         --repositories     # Include repository imports and RepositoryMixin
         --services         # Include service-proxy imports and ServiceMixin
         --apiconnect       # Include ApiConnectComponent
         --description      # Description for the application
         --outdir           # Project root directory for the application
         --eslint           # Enable eslint
         --prettier         # Enable prettier
         --mocha            # Enable mocha
         --loopbackBuild    # Use @loopback/build
         --vscode           # Use preconfigured VSCode settings
         --private          # Mark the project private (excluded from npm publish)
  -c,    --config           # JSON file name or value to configure options
  -y,    --yes              # Skip all confirmation prompts with default or provided value
         --format           # Format generated code using npm run lint:fix
  -pm,   --packageManager   # Change the default package manager                            Default: npm

Arguments:
  name  # Project name for the application  Type: String  Required: false

➜  loopback-next git:(changes-in-cli) 
```

Comment (raymondfeng) [reply packages/cli/test/integration/generators/copyright-monorepo.integration.js]: I was able to fix the issue and pushed a commit back to your branch. Please note it's also rebased against latest master. You will need to reset your local checkout.

Comment (frbuceta) [reply packages/cli/test/integration/generators/copyright-monorepo.integration.js]: Done. Thanks for helping me

**Comment by frbuceta** [comment]
I just restarted testing and it gave me this error

```
$ if [[ "${TRAVIS_OS_NAME}" == "linux" ]] && [[ $(echo $TRAVIS_NODE_VERSION | cut -d'.' -f1) == "10" ]]; then npm run coverage:ci; fi
> loopback-next@0.1.0 coverage:ci /home/travis/build/strongloop/loopback-next
> node packages/build/bin/run-nyc report --reporter=text-lcov | coveralls
/home/travis/build/strongloop/loopback-next/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 422 {"message":"service_job_id (325711990) must be unique for Travis Jobs not supplying a Coveralls Repo Token","error":true}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! loopback-next@0.1.0 coverage:ci: `node packages/build/bin/run-nyc report --reporter=text-lcov | coveralls`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the loopback-next@0.1.0 coverage:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-04-30T14_00_11_611Z-debug.log
Done. Your build exited with 0.
```

File: packages/cli/generators/app/index.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ module.exports = class AppGenerator extends ProjectGenerator {
 
   async setOptions() {
     await super.setOptions();
-    if (this.shouldExit()) return;
+    if (this.shouldExit()) return false;
```
Comment (raymondfeng) [inline packages/cli/generators/app/index.js:62]: This change is not needed. Let's revert to make the diff cleaner.

File: packages/cli/lib/project-generator.js [hunk 4]
```diff
@@ -217,13 +217,35 @@ module.exports = class ProjectGenerator extends BaseGenerator {
...
+  promptYarnInstall() {
+    if (this.shouldExit()) return false;
+    const prompts = [
+      {
+        type: 'confirm',
+        name: 'yarn',
+        message: g.f('Yarn is available. Do you prefer to use it by default?'),
+        when:
+          !this.options.packageManager &&
+          this.spawnCommandSync('which', ['yarn'], {stdio: false}).status === 0,
```
Comment (raymondfeng) [inline packages/cli/lib/project-generator.js:229]: Is `which` available on Windows? If not, maybe the following is better:

```js
this.spawnCommandSync('yarn', ['help'], {stdio: false}).status === 0,
```

**Comment by raymondfeng** [comment]
We should add a travis task to run the CLI acceptance test such as https://github.com/strongloop/loopback-next/blob/master/packages/cli/test/acceptance/app-run.acceptance.js with `yarn`. Let's do it in a follow-up PR.

**Comment by raymondfeng** [comment]
@frbuceta Thank you for the great efforts. Please squash commits to ensure clean history.

File: packages/cli/generators/app/index.js (hunk 3 above)
Comment (frbuceta) [reply packages/cli/generators/app/index.js:62]: Done

File: packages/cli/lib/project-generator.js (hunk 4 above)
Comment (frbuceta) [reply packages/cli/lib/project-generator.js:229]: well seen

**Comment by frbuceta** [comment]
Done

**Comment by frbuceta** [comment]
> We should add a travis task to run the CLI acceptance test such as https://github.com/strongloop/loopback-next/blob/master/packages/cli/test/acceptance/app-run.acceptance.js with `yarn`. Let's do it in a follow-up PR.

I'll get on with it as soon as I merge this PR

**Pull request merged**